include/votca/xtp/threecenter.h

```diff
@@ -115,27 +110,28 @@ namespace votca {
                 return _nmax;
             }
 
-            int get_mtot() const {
+            int msize() const {
                 return _mtotal;
             }
 
-            int get_ntot() const {
+            int nsize() const {
                 return _ntotal;
             }
 
 
-            void Initialize(int _basissize, int mmin, int mmax, int nmin, int nmax);
+            void Initialize(int basissize, int mmin, int mmax, int nmin, int nmax);
 
-            void Prune(int min, int max);
-            void Print(std::string ident);
             void Fill(const AOBasis& auxbasis, const AOBasis& dftbasis, const Eigen::MatrixXd& dft_orbitals);
+            //Rebuilds ThreeCenterIntegrals, only works if the original basisobjects still exist
+            void Rebuild(){
+                Fill(*_auxbasis,*_dftbasis,*_dft_orbitals);
+            }
 
             void MultiplyRightWithAuxMatrix(const Eigen::MatrixXd& AuxMatrix);
 
-            void Cleanup();
```

Comment:
A naive question, why is this cleanup method remove?

Reply:
no not naive, a lot of the classes have still methods like free matrix or cleanup. Thiis normally has one of two reasons a) the class has a reasonably big matrix in it which takes up space b) Something with pointers. Now the second case should not be relevant here anymore. For the first case we have a lot shorter functions and classes now, so objects fall out of scope a lot quicker.  Secondly, our systems are larger now and only very few objects actually matter in size due to the scaling. Thirdly, I prefer I bit more memory and shorter code to situations where cleanup has already been called and someone accesses an invalid object